platforms/chibios/boards/BONSAI_C4/configs/config.h

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2022 David Hoelscher, customMK
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+// Bonsai C4 includes Vbus sensing; derived designs that use PA9 for other purposes
+// may disable Vbus sensing with #define BOARD_OTG_NOVBUSSENS 1
+
+#ifndef EARLY_INIT_PERFORM_BOOTLOADER_JUMP
+#    define EARLY_INIT_PERFORM_BOOTLOADER_JUMP TRUE
+#endif
+
+// FRAM configuration
+#ifndef EXTERNAL_EEPROM_SPI_SLAVE_SELECT_PIN
+#    define EXTERNAL_EEPROM_SPI_SLAVE_SELECT_PIN PAL_LINE(GPIOA, 0)
+#    define EXTERNAL_EEPROM_SPI_CLOCK_DIVISOR 8 // 96MHz / 8 = 12MHz; max supported by MB85R64 is 20MHz
+#    define EXTERNAL_EEPROM_BYTE_COUNT 8192
+#    define EXTERNAL_EEPROM_PAGE_SIZE 64 // does not matter for FRAM, just sets the RAM buffer size in STM32F chip
+#    define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 8191
+#endif
+
+// External flash configuration
+#ifndef EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN
+#    define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN PAL_LINE(GPIOB, 12)
+#    define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 1  // 96MHz; max supported by W25Q128JV is 133MHz
```

Comment:
Minimum clock divisor on STM32 is `/2` -- it'd work with this code in place, as it'll default to `/2` anyway, but for clarity it should probably be fixed up.
```suggestion
#    define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 2  // 48MHz; max supported by W25Q128JV is 133MHz
```